CRM/Contribute/BAO/Contribution.php

```diff
@@ -4597,4 +4597,48 @@ public static function recordPartialPayment($contribution, $params) {
     return CRM_Core_BAO_FinancialTrxn::create($balanceTrxnParams);
   }
 
+  /**
+   * Function to add payments for contribution
+   * for Partially Paid status
+   *
+   * @param array $lineItems
+   * @param array $contributions
+   *
+   */
+  public static function addPayments($lineItems, $contributions) {
+    foreach ($contributions as $k => $contribution) {
+      if ($contribution->contribution_status_id != CRM_Core_OptionGroup::getValue('contribution_status', 'Partially paid', 'name')) {
+        continue;
+      }
+      // get financial trxn which is a payment
+      $sql = "SELECT ft.id 
+      FROM civicrm_financial_trxn ft 
+      INNER JOIN civicrm_entity_financial_trxn eft ON eft.financial_trxn_id = ft.id AND eft.entity_table = 'civicrm_contribution'
+      WHERE eft.entity_id = %1 AND ft.is_payment = 1";
+      $ftId = CRM_Core_DAO::singleValueQuery($sql, array(1 => array($contribution->id, 'Integer')));
+      // get financial item
+      $sql = "SELECT fi.id, li.price_field_value_id
+      FROM civicrm_financial_item fi
+      INNER JOIN civicrm_line_item li ON li.id = fi.entity_id
+      WHERE li.contribution_id = %1";
+      $dao = CRM_Core_DAO::executeQuery($sql, array(1 => array($contribution->id, 'Integer')));
+      while ($dao->fetch()) {
+        $ftIds[$dao->price_field_value_id] = $dao->id;
+      }
+      foreach ($lineItems as $key => $value) {
+        $paid = $value['line_total'] * ($contribution->net_amount / $contribution->total_amount);
+        // Record Entity Financial Trxn
+        $params = array(
+          'entity_table' => 'civicrm_financial_item',
+          'financial_trxn_id' => $ftId,
+          'amount' => $paid,
+          'entity_id' => $ftIds[$value['price_field_value_id']],
+        );
+        $entityTrxn = new CRM_Financial_DAO_EntityFinancialTrxn();
```

Comment:
You should really use the BAO create or api method and avoid save unless you have a compelling reason to use it.

Save bypasses the hooks - which in this case you can probably get away with because this particular entity doesn't call hooks. But you shouldn't assume that will remain the case - if someone later adds a hook to the function then it should apply to all instances where it is called.

You might like to try adding the test & then ensuring that swapping this to create doesn't cause any change - things like that are great to do in conjunction with test-writing